File: leetcode_JS/str_Substring.js
/*
https://leetcode.com/problems/implement-strstr/description
Return the index of the first occurrence of needle in haystack, or -1 if needle is not part of haystack.
*/

var strStr = function(haystack, needle) {
    if (needle === '') { return 0; }
    for (let i=0 ; i<haystack.length ; i++) {
        if (haystack[i] === needle[0]) {
            substr = haystack.substr(i, needle.length);
            if (needle === substr) { return i; }
        }
    }
    return -1;
};

console.log(strStr("mississippi", "issip")); // 4

File: leetcode_Python/str_Substring.py
'''
https://leetcode.com/problems/implement-strstr/description
'''

def strStr(haystack, needle):
    if needle == '': return 0
    for i in range(len(haystack)):
        if haystack[i] == needle[0]:
            if haystack[i:i+len(needle)] == needle:
                return i
    return -1

print(strStr('hello', 'll')) # 2
print(strStr('asdf', '')) # 0
